packages/fastify-renderer/src/node/renderers/react/ReactRenderer.tsx

```diff
@@ -69,14 +58,111 @@ export class ReactRenderer implements Renderer {
       for (const renderable of renderables) {
         await this.loadModule(this.entrypointRequirePathForServer(renderable))
       }
+
+      const mode = this.options.mode
+      this.transformHooks = this.plugin.hooks
+        .map(unthunk)
+        .map((hook) => hook.transform?.absolutePath)
+        .flatMap((hook) => (hook ? [hook] : []))
```

Comment:
tiny nit: this allocates a bunch of extra arrays, you can use the `compact` function from `lodash` if you want instead

Reply:
It does, but it only runs once during the `prepare` stage. Inconsequential imo. 

Pulling in `lodash` seems like a bigger deal.

This is just for [type-safety](https://stackoverflow.com/a/59726888/4088472), I can just change it to a cast if you rather avoid using `flatMap` for this.

Reply:
The way it reads to me is that it filters out undefined values from the array, that's more than just type safety no? There can be hooks which dont provide a transform, no?

Reply:
> The way it reads to me is that it filters out undefined values from the array, that's more than just type safety no?

Yes, but its a work-around a TS "issue". 

`array.filter((value: T | null) => !!value)` works to remove `null` but TypeScript doesn't understand it.

This would work too:

```
array.filter((s): s is string => Boolean(s))
```

I'll switch to it. I think it's far less objectionable.